ThreeBotPackages/threefold/tfgrid_directory/chatflows/register_farm.py

```diff
@@ -0,0 +1,53 @@
+from Jumpscale import j
+
+
+def chat(bot):
+    explorer = j.clients.gedis.get(name="explorer", port=8901, host="explorer.testnet.grid.tf")
+
+    question = "Please enter your farm name: {}"
+    remark = ""
+    while True:
+        name = bot.string_ask(question.format(remark))
+        try:
+            found = explorer.actors.farms.get(name=name)
+        except j.exception.NotFound:
+            found = False
+        if found:
+            remark = "(this farm name  is already used, please choose another one)"
+            continue
+        else:
+            break
+
+    question = "Please enter your email: {}"
+    remark = ""
+    while True:
+        email = bot.string_ask(question.format(remark))
```

Comment:
we used to have a validate parameter, please use it instead `email = bot.string_ask("Enter email", validate={"required": True, "email": True})`

Reply:
is there a way to pass a custom function in the validate argument ?

Reply:
The client ends up calling the functions here, so may be you can add a function there instead of passing one?
https://github.com/threefoldtech/jumpscaleX_threebot/blob/development/ThreeBotPackages/threebot/chat/static/chat/js/validators.js 
and this is the 3bot client https://github.com/threefoldtech/jumpscaleX_threebot/blob/development/ThreeBotPackages/threebot/chat/static/chat/js/bot_client.js